pkg/machine/e2e/init_test.go

```diff
@@ -98,6 +99,59 @@ var _ = Describe("podman machine init", func() {
 		}
 	})
 
+	It("run playbook", func() {
+		str := randomString()
+
+		// ansible playbook file to create a text file containing a random string
+		playbookContents := fmt.Sprintf(`
+- name: Simple podman machine example
+  hosts: localhost
+  tasks:
+    - name: create a file
+      ansible.builtin.copy:
+        dest: /home/core/foobar.txt
+        content: "%s\n"
+`, str)
+
+		tmpDir, err := os.MkdirTemp("", "")
+		defer func() { _ = utils.GuardedRemoveAll(tmpDir) }()
+		Expect(err).ToNot(HaveOccurred())
+
+		// create the playbook file
+		playbookFile, err := os.Create(filepath.Join(tmpDir, "playbook.yaml"))
+		Expect(err).ToNot(HaveOccurred())
+
+		// write the desired contents into the file
+		_, err = playbookFile.WriteString(playbookContents)
+		Expect(err).To(Not(HaveOccurred()))
+
+		name := randomString()
+		i := new(initMachine)
+		session, err := mb.setName(name).setCmd(i.withImage(mb.imagePath).withRunPlaybook(filepath.Join(tmpDir, "playbook.yaml")).withNow()).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		// calculate sha256sum of local playbook file
+		cmd := exec.Command("sha256sum", filepath.Join(tmpDir, "playbook.yaml"))
+		shasum, err := cmd.Output()
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
I doubt this will work on windows.

I think you should just calc the hash directly in the go code. However given the file is so small I would just compare the full output of the file and not the hash. If there is a small difference in the file, i.e. newline the hash difference is totally useless for debugging this.

As such I would just `machine ssh cat playbook`
and then out `Expect(sshSession.outputToString()).To(Equal(playbookContents))`